frontend/src/components/question-creation/question-elements/modals/multiple-choice/AddMultipleChoiceModal.tsx

```diff
@@ -15,26 +15,32 @@ const AddMultipleChoiceModal = (): React.ReactElement => {
   const {
     showAddMultipleChoiceModal,
     setShowAddMultipleChoiceModal,
+    isMultiSelect,
   } = useContext(QuestionEditorContext);
   const closeModal = () => {
     setShowAddMultipleChoiceModal(false);
     onClose();
   };
 
+  const multiData = isMultiSelect
+    ? { type: QuestionElementType.MULTI_SELECT }
+    : { type: QuestionElementType.MULTIPLE_CHOICE };
+
   const { setQuestionElements } = useContext(QuestionEditorContext);
   const addMultipleChoiceElement = (data: MultipleChoiceData) => {
     setQuestionElements((prevElements) => [
       ...prevElements,
       {
         id: uuidv4(),
-        type: QuestionElementType.MULTIPLE_CHOICE,
+        ...multiData,
         data,
       },
     ]);
   };
 
   return (
     <MultipleChoiceModal
+      isMultiSelect={isMultiSelect}
       isOpen={showAddMultipleChoiceModal}
```

Comment:
I think instead you can do something like `isMultipleChoice ? showAddMultipleChoiceModal : showAddMultiSelectModal`